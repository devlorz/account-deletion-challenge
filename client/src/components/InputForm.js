import PropTypes from 'prop-types'
import React from 'react'

const InputForm = ({ stack, canComment, placeHolder, isRender }) => {
  const prefill = placeHolder && canComment ? placeHolder : ''
  return !isRender ? null : (
    <div style={!canComment ? { display: 'none' } : null}>
      <input type="text" name={stack} placeholder={prefill} />
    </div>
  )
}

InputForm.PropTypes = {
  stack: PropTypes.string,
  canComment: PropTypes.bool,
  placeHolder: PropTypes.string,
  isRender: PropTypes.bool,
}

export default InputForm
