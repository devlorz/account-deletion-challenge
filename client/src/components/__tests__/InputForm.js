import React from 'react'
import { render } from 'react-testing-library'

import InputForm from '../InputForm'

test('snapshot', () => {
  const { container } = render(
    <InputForm
      stack={'stack'}
      canComment={true}
      placeHolder={'placeholder'}
      isRender={true}
    />
  )
  expect(container.firstChild).toMatchSnapshot()
})
