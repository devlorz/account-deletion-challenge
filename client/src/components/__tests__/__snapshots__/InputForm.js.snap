// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  data-reactroot=""
>
  <input
    name="stack"
    placeholder="placeholder"
    type="text"
  />
</div>
`;
